cranelift/codegen/src/isa/x64/mod.rs

```diff
@@ -92,15 +92,16 @@ impl MachBackend for X64Backend {
     }
 
     fn unsigned_add_overflow_condition(&self) -> IntCC {
-        // Unsigned `>=`; this corresponds to the carry flag set on x86, which happens on
-        // overflow of an add.
-        IntCC::UnsignedGreaterThanOrEqual
+        // Unsigned `<`; this corresponds to the carry flag cleared on x86,
+        // which indicates an add has not overflowed.
+        IntCC::UnsignedLessThan
```

Comment:
I think the initial comment was correct since the intent of this function is to return the condition code used when there *is* an unsigned overflow, but what the function returned was not correct (my bad!).

Confirming the reasoning:
- we codegen `if !unsigned_add_overflow_condition { jmp after trap }; trap`]
- unsigned add overflow means the carry flag is set to 1, so CF = 1
- hence we want to codegen a jump when CF is 0, so `JNB`, NB is not-below
- so the condition to return in this function is the inverse of NB, that is B "below", i.e. "unsigned less" in x86-speak \o/

Reply:
It took me far too long staring at this (very thorough, thank you!) sequence of steps to internalize it; yes, you're right, and the key bit is that overflow -> carry set -> B / ult. Thanks, updated comment!